### PR TITLE
add .npy reader to builtin reader

### PR DESCRIPTION
**Pull request by thanushipeiris**
# Description
This PR addresses #2976

### CODE REVIEW

**Review by jni — APPROVED** [review]
Lovely, thanks @thanushipeiris! Could you add a test for this? It should be in the same file as and modeled after this test:

https://github.com/napari/napari/blob/023a91139060db8036862b0007bcc151b143544b/napari/plugins/_tests/test_reader_plugins.py#L13

(ie in `napari/plugins/_tests/test_reader_plugins.py`).

**Review by sofroniewn — APPROVED** [review]
LGTM too! Agreed test would be great, thanks!

**Comment by sofroniewn** [comment]
Hmm test is failing, not quite sure why - maybe @tlambert03 would see based on a quick look

**Comment by tlambert03** [comment]
> Hmm test is failing, not quite sure why - maybe @tlambert03 would see based on a quick look

because `utils.io.imsave` (used in the test here to save a temporary npy file) doesn't know how to save npy files ... use np.save instead?

File: napari/plugins/_tests/test_reader_plugins.py [hunk 1]
```diff
@@ -31,6 +31,27 @@ def test_builtin_reader_plugin():
         assert np.allclose(viewer.layers[0].data, data)
 
 
+def test_builtin_reader_plugin_npy():
+    """Test the builtin reader plugin reads a temporary npy file."""
+
+    with NamedTemporaryFile(suffix='.npy', delete=False) as tmp:
+        data = np.random.rand(20, 20)
+        utils.io.imsave(tmp.name, data)
```
Comment (sofroniewn) [inline napari/plugins/_tests/test_reader_plugins.py]:
```suggestion
        np.save(tmp.name, data)
```
from @tlambert03

**Comment by sofroniewn** [comment]
> because utils.io.imsave (used in the test here to save a temporary npy file) doesn't know how to save npy files ... use np.save instead?

Thanks @tlambert03 that did the trick. I will merge this now so I can include it in the next rc, thanks for the contribution @thanushipeiris!!

**Pull request merged**